### PR TITLE
chore(nomenclature): phrase non production components as beta

### PR DESCRIPTION
**Pull request by seanforyou23**
**What**: Steps toward closing https://github.com/patternfly/patternfly-react/issues/3624

This PR renames "experimental" things to "beta" so that the intent of the in-progress status is more clearly signaled to users.

### CODE REVIEW

File: packages/patternfly-4/react-core/src/beta/components/Drawer/examples/Drawer.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ class SimpleDrawer extends React.Component {
 
 ```js title=Basic-inline
 import React, { ReactFragment } from 'react';
-import { Drawer, DrawerPanelContent, DrawerContent } from '@patternfly/react-core/dist/esm/experimental';
+import { Drawer, DrawerPanelContent, DrawerContent } from '@patternfly/react-core/dist/esm/beta';
```
Comment (dgutride) [inline packages/patternfly-4/react-core/src/beta/components/Drawer/examples/Drawer.md]: Can these be changed to @patternfly/react-core directly now?

File: packages/patternfly-4/react-docs/gatsby-browser.js [hunk 2]
```diff
@@ -16,9 +16,9 @@ import '@patternfly/patternfly/patternfly.css';
 // Utilities
 import '@patternfly/patternfly/patternfly-addons.css';
 
-// Experimental components
+// Beta components
```
Comment (dgutride) [inline packages/patternfly-4/react-docs/gatsby-browser.js]: Are these just temporarily necessary until a core change is made to include css in the main package?

File: packages/patternfly-4/react-docs/gatsby-config.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ module.exports = {
             { section: 'charts' },
             { section: 'components' },
             { section: 'demos' },
-            { section: 'experimental' },
+            { section: 'beta' },
```
Comment (dgutride) [inline packages/patternfly-4/react-docs/gatsby-config.js]: This should go away - all beta components will be listed with the others - just noted differently in the documentation.  If you'd rather do that in a separate PR (or have somebody else make that change, it's fine with me).

File: packages/patternfly-4/react-styled-system/README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # @patternfly/react-styled-system
 
-## Warning: the styled system components are experimental. Use at your own risk!
+## Warning: the styled system components are beta. Use at your own risk!
```
Comment (dgutride) [inline packages/patternfly-4/react-styled-system/README.md]: These actually truly are experimental (and I think we should deprecate and remove) - can we leave the wording as-is?

File: packages/react-codemods/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # @patternfly/react-codemods
 
-_PatternFly React codemods are currently experimental only._
+_PatternFly React codemods are currently beta only._
```
Comment (dgutride) [inline packages/react-codemods/README.md]: These are also truly experimental - maybe we should leave it this way for now?

File: packages/patternfly-4/react-core/src/beta/components/Drawer/examples/Drawer.md (hunk 1 above)
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/beta/components/Drawer/examples/Drawer.md]: I agree for new components. I actually think we may want to still export from experimental (and from react-core) for existing components and remove that with the breaking change release.

Comment (dgutride) [reply packages/patternfly-4/react-core/src/beta/components/Drawer/examples/Drawer.md]: I agree - for this, I'm just talking about the import statements we are using - I'd rather that we just treat these components like other components to indicate the commitment to them.    That does remind me that I need to look to see if the experimental/index.js still exists...

Comment (seanforyou23) [reply packages/patternfly-4/react-core/src/beta/components/Drawer/examples/Drawer.md]: Ok think I see what you mean, pushed some changes - let me know if I missed anything!

**Comment by seanforyou23** [comment]
I've pushed some updates. We now display previously "experimental" components alongside all production components and visually mark them as "Beta" with an inline alert. This alert message needs to be reviewed.

Screen Shot 2020-02-07 at 12 35 17 PM

In addition to now having the notion of beta components, we also have support for beta properties/features on production components. No beta properties exist yet, but below is a screenshot of how they are currently wired up to display. Notice the property name is `exampleBetaProperty` and the visual treatment is to simply prefix the property name with `Beta: `, just like we do for deprecated properties. We could style it some special way, or instead just generate a new "beta" column, but I figured a first pass would be good to follow the existing convention. #3581, which is in-progress, will put this new feature to use for the first time, I'm happy to address any changes now or later to this part of the experience.

Screen Shot 2020-02-07 at 12 26 20 PM

File: packages/patternfly-4/react-docs/gatsby-browser.js (hunk 2 above)
Comment (seanforyou23) [reply packages/patternfly-4/react-docs/gatsby-browser.js]: Just waiting on these last few [changes from core](https://github.com/patternfly/patternfly-next/commit/f24376e58d941420c477e0b0aa29f985af5111ae) to come in so we can delete the rest, until they come through we still need to manually import a couple of these

**Comment by seanforyou23** [comment]
Thanks @mcarrano - checking on the styles, think we just need to pull in a recent change from core. Good catch.

As far as using a badge to flag beta properties, it shouldn't be a problem. I will take a look!

**Comment by seanforyou23** [comment]
Thx for the feedback @mcarrano - I pulled in the latest changes from core and that seems to have fixed the styles on the Data toolbar/Drawer beta component pages. Also, made some changes in an open PR in [org](https://github.com/patternfly/patternfly-org/pull/1682) that should better reflect the design @rachael-phillips provided for beta styles (sry, didn't see that before). Here's an example of what those changes across org/core produce:

Marking examples as beta:
Screen Shot 2020-02-10 at 1 14 29 PM

And then marking props as beta in the props table at the bottom: 
Screen Shot 2020-02-10 at 1 28 10 PM

Let me know if you see any issues!

File: packages/patternfly-4/react-core/src/index.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
+export * from './beta';
```
Comment (dgutride) [inline packages/patternfly-4/react-core/src/index.ts]: I'm still a bit confused about why we have a beta location - the goal was to move the components to their final place (with the other components) and just use the documentation to indicate they are beta so we wouldn't need to keep moving them.

Comment (seanforyou23) [reply packages/patternfly-4/react-core/src/index.ts]: I see, sorry I missed this! Getting that updated now
